core/src/main/scala/akka/persistence/r2dbc/snapshot/SnapshotDao.scala

```diff
@@ -0,0 +1,278 @@
+            .bind(2, persistenceId)
+
+          var bindIdx = 2
+          if (criteria.maxSequenceNr != Long.MaxValue) {
+            bindIdx += 1
+            statement.bind(bindIdx, criteria.maxSequenceNr)
+          }
+          if (criteria.minSequenceNr > 0L) {
+            bindIdx += 1
+            statement.bind(bindIdx, criteria.minSequenceNr)
+          }
+          if (criteria.maxTimestamp != Long.MaxValue) {
+            bindIdx += 1
+            statement.bind(bindIdx, criteria.maxTimestamp)
+          }
+          if (criteria.minTimestamp > 0L) {
+            bindIdx += 1
+            statement.bind(bindIdx, criteria.minTimestamp)
+          }
+          statement
+        },
+        row =>
+          Serialization.withTransportInformation(system.classicSystem.asInstanceOf[ExtendedActorSystem]) { () =>
+            deserializeSnapshotRow(row, serialization)
+          })
+      .map(_.headOption)(ExecutionContexts.parasitic)
+
+  }
+
+  def store(metadata: SnapshotMetadata, value: Any): Future[Unit] =
+    Serialization.withTransportInformation(system.classicSystem.asInstanceOf[ExtendedActorSystem]) { () =>
+      val entityTypeHint = SliceUtils.extractEntityTypeHintFromPersistenceId(metadata.persistenceId)
+      val slice = SliceUtils.sliceForPersistenceId(metadata.persistenceId, settings.maxNumberOfSlices)
+
+      val insert =
+        if (metadata.metadata.isEmpty) insertSql
+        else insertSqlMeta
+
+      val snapshot = value.asInstanceOf[AnyRef]
+      val snapshotSerializer = serialization.findSerializerFor(snapshot)
+      val snapshotManifest = Serializers.manifestFor(snapshotSerializer, snapshot)
+      val serializedSnapshot = snapshotSerializer.toBinary(snapshot)
+
+      val serializedMeta: Option[(Array[Byte], Int, String)] = metadata.metadata.map { meta =>
+        val metaRef = meta.asInstanceOf[AnyRef]
+        val metaSerializer = serialization.findSerializerFor(metaRef)
+        val metaManifest = Serializers.manifestFor(metaSerializer, metaRef)
+        (metaSerializer.toBinary(metaRef), metaSerializer.identifier, metaManifest)
+      }
+
+      r2dbcExecutor
+        .updateOne(s"insert snapshot [${metadata.persistenceId}], sequence number [${metadata.sequenceNr}]") {
+          connection =>
+            val statement =
+              connection
+                .createStatement(insert)
+                .bind(0, slice)
+                .bind(1, entityTypeHint)
+                .bind(2, metadata.persistenceId)
+                .bind(3, metadata.sequenceNr)
+                .bind(4, metadata.timestamp)
+                .bind(5, serializedSnapshot)
+                .bind(6, snapshotSerializer.identifier)
+                .bind(7, snapshotManifest)
+
+            serializedMeta.foreach { case (serializedMeta, serializerId, serializerManifest) =>
+              statement
+                .bind(8, serializedMeta)
+                .bind(9, serializerId)
+                .bind(10, serializerManifest)
```

Comment:
don't you have to bind those to null when there is no meta? see journal

Reply:
When I set null I got some error, so there's two different statements instead.